internal/resource/configmap.go

```diff
@@ -247,6 +250,43 @@ func (builder *ServerConfigMapBuilder) Update(object client.Object) error {
 		return fmt.Errorf("failed setting controller reference: %w", err)
 	}
 
+	updatedConfigMap := configMap.DeepCopy()
+	if err := removeConfigNotRequiringNodeRestart(previousConfigMap); err != nil {
+		return err
+	}
+	if err := removeConfigNotRequiringNodeRestart(updatedConfigMap); err != nil {
+		return err
+	}
```

Comment:
I'm a fan of wrapping the error with `fmt.Errorf("some context: %w", err)`, and provide some context as what we were trying to do; in this case, removing a property from the config map. Otherwise, a low level error from `Write` or `ini.Load` will bubble up and get logged in the operator logs, with no context as where this error comes from. It's a great practice to provide a bit of context around what we were trying to do, before bubbling the error.

Reply:
Good point. Done